coruscant/src/reducers/projectMembers.js

```diff
@@ -0,0 +1,55 @@
+import { normalize } from 'normalizr'
+import * as Constants from './../constants'
+import userList from './../models/userList'
+
+function makeInitialState() {
+  return {
+    contributors: {},
+    owners: {},
+    fetchedContributors: false,
+    fetchedOwners: false
+  }
+}
+
+function setProjectContributors(state, action) {
+  const normalizedContributors = normalize(action.payload.users, userList)
```

Comment:
Similar to what I was saying for the issues and statuses, this reducer should be changed to users reducer that contains all the users from the backend. Then each project will contain a list of contributor ids and owner ids. You can use those ids to find the respective user in the users object. Think about the redux store as a database. There will be a users table, projects table, build statuses table, issues table, etc. These will be mapped to each other using their ids. Take a look at https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape

Reply:
While I agree the structure you propose is certainly clean and intuitive, I think the current method meets all of our needs. I have created three reducers, projectBuilds, projectIssues, and projectMembers. These hold the fetched information about the current project. This is preferable so that we aren't fetching and storing way more information than we need. There is a tradeoff between reloading data and already having the data; however, I dont foresee this being a noticeable drawback. Overall as long as this meets our needs I don't think we should spend time restructuring the store.